lightning/src/ln/channelmanager.rs

```diff
@@ -4262,9 +4271,23 @@ where
 		}
 		let mut peer_state_lock = peer_state_mutex_opt.unwrap().lock().unwrap();
 		let peer_state = &mut *peer_state_lock;
+
+		let mut unfunded_channels = 0;
+		let best_block_height = self.best_block.read().unwrap().height();
+		for (_, chan) in peer_state.channel_by_id.iter() {
+			if chan.get_funding_tx_confirmations(best_block_height) == 0 && chan.minimum_depth() != 0 {
```

Comment:
IIUC, it means 0-conf channels are discarded from the `MAX_UNFUNDED_CHANS_PER_PEER`. Still, I think we're allocating per-peer state for them, at least `ChannelMonitor` resources. I lean to think we might have a `MAX_0CONF_UNFUNDED_CHANS_PER_PEER=100`.  Like if you have more 100 channels with the same (random) peer, it's clearly an anomaly there is no economic rationality for this, or at least you should "allowlist" this peer.

Reply:
Yes I got the idea but it doesn't seem to me the same _type_ of trust, in the sense you expect the 0conf risk to be partitioned per-channel opened with such peers. The concern is this 0conf trust being escalated to trigger a DoS hammering your _non_ 0conf channels safety. Like you might have additional checks at the application-level limiting your 0conf exposure, of which an exploited DoS will bypass.

Reply:
Sure -- Preventing block data from being received might have safety issue. From my viewpoint, the substantial difference lies in _contamination_ risk to non 0conf channels, of which I would be concerned as a a node operator. Of course, this is ultimately dependent on the 0conf counterparties selection you're using (manual based on service operator reputation, or automated based on `option_zeroconf` feature bit). 

I think the risk qualification ultimately is beyond the pure `rust-lightning` tree scope (e.g dependent on your 0conf selection heuristics) so best we can do is document better our interfaces. Tracked with https://github.com/lightningdevkit/rust-lightning/issues/1995